content/doc/developer/plugin-development/build-process.adoc

```diff
@@ -0,0 +1,28 @@
+---
+layout: developersection
+title: Plugin Build Process
+references:
+- url: ../updating-parent/
+  title: Guide to Updating Your Maven Parent POM
+- url: https://github.com/jenkinsci/plugin-pom/
+  title: Plugin parent POM 2.x and up
+- url: https://github.com/jenkinsci/maven-hpi-plugin/
+  title: Maven HPI Plugin
+---
+
+Most Jenkins plugins are built with link:https://maven.apache.org[Maven]. The link:https://github.com/jenkinsci/maven-hpi-plugin/[Maven HPI plugin] does the heavy lifting, such as bundling plugins in the HPI/JPI archive format, or running a debug Jenkins instance with the plugin.
+
+Jenkins plugins typically use the Jenkins plugin parent POM, which provides a sensible default configuration for the build.
+Up to Jenkins 1.645, the plugin POM was kept in sync with Jenkins releases, so that the minimum required Jenkins version for a plugin determined the versions of the tools used to build the plugin.
+As of Jenkins 1.646, they've been decoupled:
+Plugin POM 2.0 and newer allow specifying the core version dependency independent of its version, so that even plugins compatible with older Jenkins releases can benefit from fixes and improvements in the parent POM.
+
+== Keeping up to date
+
+It is generally recommended to continually update to recent plugin parent POMs for multiple reasons:
+
+* Plugin developers and users benefit from bug fixes and improvements to the parent POM and the tools it specifies
+* Additions to the default build like static code analysis and general tests aim to improve the overall quality of plugins.
+// TODO Need a good reference for that before including it as example:
+// For example, plugins depending on the plugins parent POM 1.596 (with maven-hpi-plugin 1.110) or newer will need to specify the <code>escape-by-default</code> in all Jelly files for the InjectedTest to pass. Plugins depending on older releases may have hidden XSS vulnerabilities.
+* Recent versions of the tools (inherited from the parent plugins POM) allow developers to use more advanced tools like PCT (plugin compatibility tester) to determine whether their plugin is compatible with newest Jenkins releases.
```

Comment:
Is there a good PCT link we can add?

Reply:
After my recent patches to PCT landing is good enough: https://github.com/jenkinsci/plugin-compat-tester . I am about extending this section later